hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -269,7 +272,12 @@ private void processRecords(ManagedRocksIterator iterator,
     // Count number of keys printed so far
     long count = 0;
     List<Future<Void>> futures = new ArrayList<>();
-    while (withinLimit(count) && iterator.get().isValid() && !exception) {
+    boolean reachedEnd = false;
+    while (withinLimit(count) && iterator.get().isValid() && !exception && !reachedEnd) {
+      // if invalid endKey is given, it is ignored
+      if (null != endKey && Arrays.equals(iterator.get().key(), getValueObject(dbColumnFamilyDef, endKey))) {
+        reachedEnd = true;
```

Comment:
Is it also possible to display a message to let user know if endKey is ignored?
Maybe something like,
`err().println("endKey: " + endKey + " provided is invalid");`
or incase endKey exceeds total number of keys!

Reply:
if endKey is not of format as per key specification, it is throwing invalid exception while convert to byte array in getValueObject() method. So its reported.
But otherwise, there is no way to define if endKey is invalid or not or it may be still way far in iteration for the limit. So this is not required.